crates/api_models/src/payments.rs

```diff
@@ -366,6 +436,58 @@ pub struct AmountDetails {
     tax_on_surcharge: Option<MinorUnit>,
 }
 
+#[cfg(feature = "v2")]
+pub struct AmountDetailsSetter {
```

Comment:
Why this intermidiate type is required?

Reply:
In order to construct a new AmountDetails api model from domain model. AmountDetailsSetter will be a replica of AmountDetails but fields will be public.

Reply:
We can come up with a better approach to eliminate AmountDetailsSetter later